PostFreeSurfer/PostFreesurferPipelineLongPrep.sh

```diff
@@ -0,0 +1,433 @@
+$log_ToolName: takes cross-sectional pre-Freesurfer output, as well as longitudinal FreeSurfer output folders.
+Creates volumes and transforms analogous to those produced by preFreesurferPipeline, but in longitudinal template space.
+
+Usage: $log_ToolName PARAMETER...
+
+PARAMETERs are [ ] = optional; < > = user supplied value
+"
+    #automatic argument descriptions
+    opts_ShowArguments
+    
+    #do not use exit, the parsing code takes care of it
+}
+
+#arguments to opts_Add*: switch, variable to set, name for inside of <> in help text, description, [default value if AddOptional], [compatibility flag, ...]
+#help info for option gets printed like "--foo=<$3> - $4"
+
+#TSC:should --path or --study-folder be the flag displayed by the usage?
+opts_AddMandatory '--path' 'StudyFolder' 'path' "folder containing all timepoins and templates" "--path"
+#The following options
+opts_AddMandatory '--subject'   'Subject'   'subject ID' "Subject label. Note: this is distinct subject label. Use separate labels for template and timepoints"
+opts_AddMandatory '--template'  'Template'  'FS template ID' "Longitudinal template ID (same as Freesurfer long template ID)"
+opts_AddMandatory '--timepoints' 'Timepoints_cross' 'FS timepoint ID(s)' "Freesurfer timepoint ID(s). For timepoint\
+                    processing, specify current timepoint. For template processing, must specify all timepoints, '@' separated"
+opts_AddMandatory '--template_processing' 'TemplateProcessing' 'create template flag' "0 if TP processing; 1 if template processing (must be run after all TP's)"
+
+#the following options have the same meaning as in PreFreesurferPipeline.
+opts_AddMandatory '--t1template' 'T1wTemplate' 'file_path' "MNI T1w template"
+opts_AddMandatory '--t1templatebrain' 'T1wTemplateBrain' 'file_path' "Brain extracted MNI T1wTemplate"
+opts_AddMandatory '--t1template2mm' 'T1wTemplate2mm' 'file_path' "MNI 2mm T1wTemplate"
+opts_AddMandatory '--t2template' 'T2wTemplate' 'file_path' "MNI T2w template"
+opts_AddMandatory '--t2templatebrain' 'T2wTemplateBrain' 'file_path' "Brain extracted MNI T2wTemplate"
+opts_AddMandatory '--t2template2mm' 'T2wTemplate2mm' 'file_path' "MNI 2mm T2wTemplate"
+opts_AddMandatory '--templatemask' 'TemplateMask' 'file_path' "Brain mask MNI Template"
+opts_AddMandatory '--template2mmmask' 'Template2mmMask' 'file_path' "Brain mask MNI 2mm Template"
+opts_AddMandatory '--fnirtconfig' 'FNIRTConfig' 'file_path' "FNIRT 2mm T1w Configuration file"
+
+#needed to transform Freesurfer mask to MNI space.
+opts_AddMandatory '--freesurferlabels' 'FreeSurferLabels' 'file' "location of FreeSurferAllLut.txt"
+opts_ParseArguments "$@"
+
+#display the parsed/default values
+opts_ShowValues
+
+# Naming Conventions
+T1wImage="T1w"
+T1wImageBrainMask="brainmask_fs"
+T1wFolder="T1w" #Location of T1w images
+T2wImage="T2w"
+T2wFolder="T2w" #Location of T2w images
+Modalities="T1w T2w"
+MNI_08mm_template="$HCPPIPEDIR/global/templates/MNI152_T1_0.8mm.nii.gz"
```

Comment:
Why do we have a variable name that says "08mm"?  The variable name should probably say "highVolRes" or something instead, and probably be set via an option, so that if someone acquires different structural resolution it can do the right thing.

Reply:
Right this could be 0.7mm or 1mm depending upon the inputs.  In the future there could be more varieties.

Reply:
Ok